api/views/workflow.py

```diff
@@ -40,10 +41,75 @@ class SourceViewSet(viewsets.ReadOnlyModelViewSet):
     ordering = ('id', )
     ordering_fields = ('long_title', )
     serializer_class = SourceSerializer
+    permission_classes = [ReadOnlyOrTokenHasScopeOrIsAuthenticated, ]
 
     def get_queryset(self):
         return Source.objects.exclude(icon='').exclude(is_deleted=True)
 
+    def create(self, request, *args, **kwargs):
+        try:
+            home_page = request.data['home_page']
+            long_title = request.data['long_title']
+            # icon = request.data['icon']
+        except KeyError as e:
+            raise ValidationError('{} is a required attribute.'.format(e))
+
+        label = '.'.join(reversed(home_page.split('//')[1].split('.')))
+
+        user_serializer = ShareUserSerializer(
+            data={'username': label, 'is_trusted': True},
+            context={'request': request}
+        )
+
+        if user_serializer.is_valid(raise_exception=True):
+
+            user_instance = user_serializer.save()
+
+            source_instance = Source(
+                user_id=user_instance.id,
+                long_title=long_title,
+                home_page=home_page,
+                name=label
+            )  # icon=icon
+            source_instance.save()
+
+            source_config_instance = SourceConfig(source_id=source_instance.id, label=label)
+            source_config_instance.save()
+
+            return Response(
```

Comment:
It's possible to render a top level with attributes and a related with id and type using [included](http://django-rest-framework-json-api.readthedocs.io/en/stable/usage.html#included) in the serializer. I couldn't get attributes from multiple models to be serialized. Using include on Source would look like:
```json
{
    "data": {
        "type": "Source",
        "id": "1B50E1-C0A-161",
        "attributes": {
            "name": "test_3",
            "homePage": null,
            "longTitle": "Test 3",
            "icon": "/icons/test_3.ico"
        },
        "relationships": {
            "user": {
                "data": {
                    "type": "ShareUser",
                    "id": "1B0D4-553-851"
                }
            }
        }
    }
}
```